src/conduit/migrations/env.py

```diff
@@ -42,10 +43,15 @@ def run_migrations_online():
     In this scenario we need to create an Engine
     and associate a connection with the context.
     """
+    from pyramid.threadlocal import get_current_registry
+
+    registry = get_current_registry()
+    allowed = {"sqlalchemy.url", "sqlalchemy.echo"}
+    settings = {
+        key: value for (key, value) in registry.settings.items() if key in allowed
+    }
```

Comment:
1. Isn't usage of `get_current_registry` seriously discouraged, especially outside of test suites?
2. Why do we need `allowed`? Security considerations or?

Reply:
1. The main reason that `get_current_registry` is discouraged, AIUI, it that it's a sign that you've designed your app wrong. If you need the registry you should have access to a request. Here we're trying to hook into the behaviour of a different tool (alembic) when it's run directly and as part of test suite setup. I don't feel bad about `get_current_registry`, myself. If you do, we can use `Base.metadata.bind.url`, but that will prevent the setting `sqlalchemy.echo` working in alembic.
2. It's because the registry's settings can contain other strings that look like sqlalchemy settings, specifically `sqlalchemy.engine`, which is an error to pass to this function.

Reply:
1. I've never used `sqlalchemy.echo`, so personally, I'd prefer `Base.metadata.bind.url`. Do you use it or know of people that use it? 
2. Makes sense.
3. One more consideration: I might be completely wrong but I think alembic takes logging configuration from `dev/prod.ini`. Which would mean that there wouldn't be any output after this PR is merged? Let me do some tests.

Reply:
I do use `sqlalchemy.echo`, but never in alembic, so I think this is fine. I'll make the change to use the Base metadata. It does make for a slightly more readable file, but I don't think we should worry about using `get_current_registry` in this kind of context if it's needed in future.